### PR TITLE
根据keyspace获取databases num时，需要加1

### LINKED ISSUE #2570: bug-Redis的数据库列表少了一个
### 重现步骤

```python

#代码文件 Archery/sql/engines/redis.py

 def get_all_databases(self, **kwargs):
        """
        获取数据库列表
        :return:
        """
        result = ResultSet(full_sql="CONFIG GET databases")
        conn = self.get_connection()
        try:
            rows = conn.config_get("databases")["databases"]
        except Exception as e:
            logger.warning(f"Redis CONFIG GET databases 执行报错，异常信息：{e}")
            dbs = [
                int(i.split("db")[1])
                for i in conn.info("Keyspace").keys()
                if len(i.split("db")) == 2
            ]
            rows = max(dbs + [16])

        db_list = [str(x) for x in range(int(rows))]
        result.rows = db_list
        return result
```

[跳转到代码行](https://github.com/hhyo/Archery/blob/v1.10.0/sql/engines/redis.py#L75)

1. 当 info Keyspace返回结果为：
db4:keys=31,expires=30,avg_ttl=918415
db33:keys=1,expires=0,avg_ttl=0
db255:keys=1,expires=0,avg_ttl=0

rows结果为255，
db_list结果为0到254

### 预期外的结果

db_list预期结果为0到255，
db_list实际为0到254，这里少了一个数字。

 

### 日志文本

_No response_

### 版本

1.10

### 部署方式

Docker

### 是否还有其他可以辅助定位问题的信息？比如数据库版本等

_No response_

### PR DESCRIPTION
**Pull request by feiazifeiazi**
fix https://github.com/hhyo/Archery/issues/2570

### CODE REVIEW

File: sql/engines/redis.py [hunk 1]
```diff
@@ -63,14 +63,19 @@ def get_all_databases(self, **kwargs):
         conn = self.get_connection()
         try:
             rows = conn.config_get("databases")["databases"]
+            rows = rows
+            rows = rows
+            rows = rows
+            rows = rows
+            rows = rows
```
Comment (LeoQuote) [inline sql/engines/redis.py]: 不可接受

**Review by LeoQuote — CHANGES_REQUESTED** [review]

**Comment by LeoQuote** [comment]
加 mock 就好了啊，你正好还可以测试下你的case

**Comment by feiazifeiazi** [comment]
> 加 mock 就好了啊，你正好还可以测试下你的case

本地单元测试时，总是报这个错。 “报auth_group表不存在。”
加的单元测试，都测不了。。好烦。

2024-04-10 18:14:34,861 - environ.environ - WARNING - Invalid line: AUTH_LDAP_ALWAYS_UPDATE_USER = True
2024-04-10 18:14:35,020 - archery.settings - INFO - 当前生效的外部认证方式：LDAP
2024-04-10 18:14:35,020 - archery.settings - INFO - 认证后端：('django_auth_ldap.backend.LDAPBackend', 'django.contrib.auth.backends.ModelBackend')
============================= test session starts =============================
platform win32 -- Python 3.11.2, pytest-8.1.1, pluggy-1.4.0
django: version: 4.1.10, settings: archery.settings (from ini)
rootdir: d:\Users\Administrator\Source\Workspaces\Archery_Genuine2
configfile: pyproject.toml
plugins: django-4.8.0, mock-3.14.0
collected 1 item

sql\engines\tests.py 
Got an error creating the test database: (1007, "Can't create database 'test_archery'; database exists")
E

=================================== ERRORS ====================================
_____________ ERROR at setup of TestRedis.test_get_all_databases ______________

self = <django.db.backends.utils.CursorWrapper object at 0x0000025B59F3F2D0>
sql = 'ALTER TABLE `sql_users_groups` ADD CONSTRAINT `sql_users_groups_group_id_d572a82e_fk_auth_group_id` FOREIGN KEY (`group_id`) REFERENCES `auth_group` (`id`)'
params = ()
ignored_wrapper_args = (False, {'connection': <DatabaseWrapper vendor='mysql' alias='default'>, 'cursor': <django.db.backends.utils.CursorWrapper object at 0x0000025B59F3F2D0>})

    def _execute(self, sql, params, *ignored_wrapper_args):
        self.db.validate_no_broken_transaction()
        with self.db.wrap_database_errors:
            if params is None:
                # params default might be backend specific.
                return self.cursor.execute(sql)
            else:
>               return self.cursor.execute(sql, params)

c:\Users\Administrator\archery_venv\Lib\site-packages\django\db\backends\utils.py:89: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _

**Review by LeoQuote — APPROVED** [review]
100% 覆盖率👏

**Pull request merged**